recipes/ceres-solver/all/conanfile.py

```diff
@@ -47,6 +66,20 @@ def config_options(self):
             del self.options.use_CXX11_threads
             del self.options.use_CXX11
 
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires("eigen/3.4.0")
+        if self.options.use_glog:
+            self.requires("glog/0.5.0")
+        if self.options.use_gflags:
+            self.requires("gflags/2.2.2")
+            self.options["gflags"].nothreads = False
```

Comment:
```suggestion
    def configure(self):
        self.options["gflags"].nothreads = False
        if self.options.shared:
            del self.options.fPIC

    def requirements(self):
        self.requires("eigen/3.4.0")
        if self.options.use_glog:
            self.requires("glog/0.5.0")
        if self.options.use_gflags:
            self.requires("gflags/2.2.2")
```

Reply:
yes it would be neater, and also check the value in validate()